src/targets/gpu/mlir.cpp

```diff
@@ -164,10 +164,10 @@ std::string mlir_print(F f, T x)
     return ss.str();
 }
 
-const std::unordered_set<std::string>& get_xdlops_archs()
+bool has_xdlops(const std::string& target_arch)
 {
-    static std::unordered_set<std::string> supported_archs{"gfx908", "gfx90a"};
-    return supported_archs;
+    const auto device_name = trim(split_string(target_arch, ':').front());
+    return (contains(device_name, "gfx9") and device_name >= "gfx908");
```

Comment:
There is no set logic on gfx arch name e.g. it could be `gfx90a` or `gfx1030`. IMO `>=` in such cases is not good check. 

is there a way to retrieve such information by calling any HIP API or by other means ?